src/Nether.Analytics.DataLake/Scheduler/DateTimeUtilities.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Nether.Analytics.DataLake
+{
+    public static class DateTimeUtilities
+    {
+        /// <summary>
+        /// Parses a new DateTime from format yyyyMMdd-HHmm
+        /// </summary>
+        /// <param name="from">DateTime in string format yyyyMMdd-HHmm</param>
+        /// <returns>A new DateTime</returns>
+        public static DateTime FromSpecialString(string from)
+        {
+            return DateTime.ParseExact(from, "yyyyMMdd-HHmm", CultureInfo.InvariantCulture);
+        }
+
+        /// <summary>
+        /// Converts a DateTime to format yyyyMMdd-HHmm
+        /// </summary>
+        /// <param name="dt">Provided DateTime</param>
+        /// <returns>A string in format yyyyMMdd-HHmm</returns>
+        public static string ToSpecialString(DateTime dt)
+        {
+            return dt.ToString("yyyyMMdd-HHmm");
+        }
+
+        /// <summary>
+        /// Calculates time periods for a set timespan and interval
+        /// For instance, if timespan is 3 hours and interval is 30 minutes, it will return 3 * 60 / 30 = 6
+        /// </summary>
+        /// <param name="ts">TimeSpan</param>
+        /// <param name="ji">Interval enumeration</param>
+        /// <returns></returns>
+        public static int CalculateTimePeriods(TimeSpan ts, JobInterval ji)
```

Comment:
Consider making this pluggable, by breaking out the calculations of time periods into a separate interface with different implementations depending on what schedule to use.